stellarphot/core.py

```diff
@@ -715,6 +800,225 @@ def catalog_name(self):
+        For example, the Johnson V magnitude column is
+        ``Vmag`` or ``V_mag``. The default value for ``mag_column_regex`` will match any
+        column name that starts with a letter or letters, followed by ``mag`` or
+        ``_mag`` with an underscore in between.
+
+        In many Vizier catalogs, the color columns are named with the passbands
+        separated by a hyphen. For example, the Johnson V-I color column is
+        ``V-I``. The default value for ``color_column_regex`` will match any
+        column name that starts with a letter or letters, followed by a hyphen,
+        followed by a letter or letters.
+        """
+
+        if isinstance(header_or_center, SkyCoord):
+            # Center was passed in, just use it.
+            center = header_or_center
+            if clip_by_frame:
+                raise ValueError('To clip entries by frame you must use '
+                                'a WCS as the first argument.')
+        else:
+            # Find the center of the frame
+            shape = (header_or_center['NAXIS2'], header_or_center['NAXIS1'])
+            center = header_or_center.wcs.pixel_to_world(shape[1] / 2,
+                                                         shape[0] / 2)
+
+        # Get catalog via cone search
+        Vizier.ROW_LIMIT = -1  # Set row_limit to have no limit
+        cat = Vizier.query_region(center, radius=radius, catalog=desired_catalog)
+
+        # Vizier always returns list even if there is only one element. Grab that
+        # element.
+        cat = cat[0]
+
+        if prepare_catalog is not None:
+            final_cat = prepare_catalog(cat)
+        else:
+            final_cat = CatalogData._tidy_vizier_catalog(cat, mag_column_regex,
+                                                         color_column_regex)
+
+        # Since we go through pandas, we lose the units, so we need to add them back
+        # in.
+        #
+        # We need to swap the key/values on the input map to get the old column names
+        # as values.
+        invert_map = {v: k for k, v in colname_map.items()}
+        final_cat[invert_map['ra']].unit = u.deg
+        final_cat[invert_map['dec']].unit = u.deg
+
+        # Make the CatalogData object....
+        cat = cls(input_data=final_cat,
+                  colname_map=colname_map,
+                  catalog_name=desired_catalog,
+                  catalog_source='Vizier')
+
+        # ...and now that the column names are standardized, clip by frame if
+        # desired.
+        if clip_by_frame:
+            cat_coords = SkyCoord(ra=cat['ra'], dec=cat['dec'])
+            x, y = header_or_center.wcs.all_world2pix(cat_coords.ra, cat_coords.dec, 0)
+            in_x = (x >= padding) & (x <= header_or_center.wcs.pixel_shape[0] - padding)
+            in_y = (y >= padding) & (y <= header_or_center.wcs.pixel_shape[1] - padding)
+            in_fov = in_x & in_y
+            cat = cat[in_fov]
+
+        return cat
```

Comment:
Logic looks OK as far as I can tell for this from_vizier method.  You have a test for it in test_core.py, so I'll trust it for now as long as the tests passes.